jgrapht-core/src/main/java/org/jgrapht/alg/intervalgraph/IntervalGraphRecognizer.java

```diff
@@ -0,0 +1,142 @@
+    List<V> sweepGamma = lexBreadthFirstSearchPlus(graph, vertexB, arrayOf(sweepBeta));
+    V vertexC = lastElementOf(sweepGamma);
+
+    // Step 4 - LBFS+ from the last vertex of the previous sweep
+    // Input - the result of previous sweep gamma, vertex c
+    // Output - the result of current sweep delta, further last vertex d visited by current sweep
+    List<V> sweepDelta = lexBreadthFirstSearchPlus(graph, vertexC, arrayOf(sweepGamma));
+    V vertexD = lastElementOf(sweepDelta);
+
+    // Additionally, calculate the index and the corresponding A set for each vertex
+
+    // Step 5 - LBFS+ from the last vertex of the previous sweep
+    // Input - the result of previous sweep delta, vertex d
+    // Output - the result of current sweep epsilon, further last vertex e visited by current sweep
+    List<V> sweepEpsilon = lexBreadthFirstSearchPlus(graph, vertexD, arrayOf(sweepDelta));
+    V vertexE = lastElementOf(sweepEpsilon);
+
+    // Additionally, calculate the index and the corresponding B set for each vertex
+
+    // Step 6 - LBFS* with the resulting sweeps
+    // Input - the result of sweep gamma and sweep epsilon
+    // Output - the result of current sweep zeta
+    List<V> sweepZeta = null; // TODO: replace by the invocation of lexBreadthFirstSearchStar
+
+    // if sweepZeta is umbrella-free, then the graph is interval.
+    // otherwise, the graph is not interval
+    return isIOrdering(arrayOf(sweepZeta), graph);
+  }
+
+  /**
+   * Calculates if the given sweep is an I-Ordering
+   * (according to the Graph graph)
+   *
+   * @param sweep the order we want to check if its an I-Order
+   * @param graph the graph we want to check if its an I-Order
+   * @return true, if sweep is an I-Order according to graph
+   */
+  private static <V> boolean isIOrdering(V[] sweep, Graph graph) {
+    for (int i = 0; i < sweep.length - 2; i++) {
+      for (int j = i + 1; j < sweep.length - 1; j++) {
+        for (int k = j + 1; k < sweep.length; k++) {
+          boolean edgeIJ = graph.containsEdge(sweep[i], sweep[j]);
+          boolean edgeIK = graph.containsEdge(sweep[i], sweep[k]);
+          if (!edgeIJ && edgeIK) {
+            return false;
+          }
+        }
+      }
+    }
+    return true;
+  }
+
+  /**
+   * return the last element of the given list
+   *
+   * @param list
+   * @param <V>  the generic type representing vertices
+   * @return
+   */
+  private static <V> V lastElementOf(List<V> list) {
+    if (list == null) {
+      throw new IllegalArgumentException("List parameter cannot be null.");
+    }
+
+    return list.get(list.size() - 1);
+  }
+
+  /**
+   * return a random element of the given set
+   *
+   * @param set
+   * @param <V> the generic type representing vertices
+   * @return
+   */
+  private static <V> V randomElementOf(Set<V> set) {
+    if (set == null) {
+      throw new IllegalArgumentException("List parameter cannot be null.");
+    }
+
+    int index = new Random().nextInt(set.size());
```

Comment:
We should store the Random object. Otherwise new Random().nextInt(...), when called a lot in a small loop, might return the same value (as it is initialized with the system time)